magento/pwa-studio
# [PWA-2400] Update packages and dependencies

## Description

Update packages and dependencies to reduce number of High+ vulnerabilities in yarn and npm audit
`yarn audit` vulnerabilities were reduced from 82 to 21
`npm audit` vulnerabilities were reduced from 78 to 17
Vulnerabilities related to Webpack 5, ESLint 8 and Storybook were out of scope as they have dedicated stories.

## Related Issue

Closes PWA-2400

## Acceptance

### Verification Stakeholders

Project maintainers

### Specification

Dependencies from root package, create-pwa, pwa-buildpack, venia-concept, venia-integration-tests are impacted

**Still generating vulnerabilities - refactor required**
`@storybook/react`: 6.3.7 / 6.4.8 (To be updated in PWA-1867)
`webpack` : 4.46.0 / 5.65.0 (To be updated to 5 in PWA-1893)
`copy-webpack-plugin`: Updated to 6.4.0 / 10.0 (7.0 and adove requires Webpack5, tbd in PWA-1893).
`boxen`: 5.1.2 / 6.2.1 (Version 6 introduces breaking changes - now pure ESM imports)
`@pmmmwh/react-refresh-webpack-plugin`: 0.4.3 / 0.5.3

**Still generating vulnerabilities - module updated to latest version**
`danger`: 10.7.1 / 10.7.1 (published a month ago)
`graphql-cli`: 4.1.0 / 4.1.0 (published a year ago)
`@graphql-cli/validate`: 2.1.0 / 2.1.0 (published 2 years ago)
`eslint-plugin-package-json`: 0.1.4 / 0.1.4 (published 2 years ago)
`tap-diff`: 0.1.1 / 0.1.1 (published 6 years ago).

**No longer generating vulnerabilities**
`eslint`: Updated to 7.32.0 / latest 8.4.1 (Breaking changes - To be updated in PWA-2451)
`jest`: Updated to 26.6.3 / latest 27.4.3 (Version 27 has multiple breaking changes)
`chokidar`: Updated to 3.5.2
`coveralls`: Updated to 3.1.1
`graphql-config`: Updated to 3.4.1
`jest-junit`: Updated to 13.0.0
`bundlesize`: Updated to 0.18.1
`inquirer`: Updated to 8.2.0
`envalid`: Updated to 6.0.1
`yargs`: Updated to 17.3.0
`sharp`: Updated to 0.29.3

## Verification Steps

#### Test scenario(s) for direct fix/feature
- Run `yarn audit` and `npm audit` at root, and verify vulnerabilities have been reduced
- Perform general regression smoke test on project and updated packages
- Verify `yarn test` works properly
- Verify that `yarn run lint` and `yarn run prettier` still works
- Create scaffold project and verify `yarn build` and `yarn start` still works

#### Is Browser/Device testing needed?

No, device and browser testing is not needed

#### Any ad-hoc/edge case scenarios that need to be considered?

Not that I can think of, but I still have limited knowledge of the full solution, so any feedback is appreciated

## Screenshots / Screen Captures (if appropriate)
![yarn audit develop](https://user-images.githubusercontent.com/25036022/145638826-8750fd87-2e63-40f2-8ce0-3ab68bb9e49b.PNG)
![yarn audit 2400](https://user-images.githubusercontent.com/25036022/145638824-a10842f5-0866-49b3-9e32-3b410de2fe40.PNG)

## Breaking Changes (if any)

Should not have any breaking change

## Checklist

-   I have added tests to cover my changes, if necessary.
-   I have added translations for new strings, if necessary.
-   I have updated the documentation accordingly, if necessary.
